examples/fluids/qfunctions/newtonian.h

```diff
@@ -710,69 +461,57 @@ CEED_QFUNCTION(IJacobian_Newtonian)(void *ctx, CeedInt Q,
 
     // Total flux
     CeedScalar dFlux[5][3];
-    for (int j=0; j<3; j++) {
-      dFlux[0][j] = dF_inviscid[j].density;
-      for (int k=0; k<3; k++)
-        dFlux[k+1][j] = dF_inviscid[j].momentum[k] - dstress[k][j];
-      dFlux[4][j] = dF_inviscid[j].E_total + dFe[j];
-    }
+    FluxTotal(dF_inviscid, dstress, dFe, dFlux);
 
-    for (int j=0; j<3; j++) {
-      for (int k=0; k<5; k++) {
+    for (int j=0; j<3; j++)
+      for (int k=0; k<5; k++)
         Grad_v[j][k][i] = -wdetJ * (dXdx[j][0] * dFlux[k][0] +
                                     dXdx[j][1] * dFlux[k][1] +
                                     dXdx[j][2] * dFlux[k][2]);
-      }
-    }
 
     const CeedScalar dbody_force[5] = {0, ds.U.density *g[0], ds.U.density *g[1], ds.U.density *g[2], 0};
     for (int j=0; j<5; j++)
       v[j][i] = wdetJ * (context->ijacobian_time_shift * dU[j] - dbody_force[j]);
 
-    if (1) {
-      CeedScalar jacob_F_conv[3][5][5] = {0};
-      computeFluxJacobian_NS(jacob_F_conv, s.U.density, s.Y.velocity, s.U.E_total,
-                             gamma, g, x_i);
-      CeedScalar grad_dU[5][3];
-      for (int j=0; j<3; j++) {
-        grad_dU[0][j] = grad_ds[j].U.density;
-        for (int k=0; k<3; k++) grad_dU[k+1][j] = grad_ds[j].U.momentum[k];
-        grad_dU[4][j] = grad_ds[j].U.E_total;
-      }
-      CeedScalar dstrong_conv[5] = {0};
-      for (int j=0; j<3; j++)
-        for (int k=0; k<5; k++)
-          for (int l=0; l<5; l++)
-            dstrong_conv[k] += jacob_F_conv[j][k][l] * grad_dU[l][j];
-      CeedScalar dstrong_res[5];
+    // -- Stabilization method: none, SU, or SUPG
+    // Strong convective flux
+    CeedScalar dstrong_conv[5] = {0};
+    FluxInviscidStrong(context, s, grad_ds, dstrong_conv);
+
+    // Strong residual
+    CeedScalar dstrong_res[5];
+    for (int j=0; j<5; j++)
+      dstrong_res[j] = context->ijacobian_time_shift * dU[j] +
+                       dstrong_conv[j] - dbody_force[j];
```

Comment:
I guess one way to write this would be to have NONE/SU/SUPG just change what terms are active on this line, and then you could remove the switch down below. The current structure is fine too.

Reply:
What do you think of [the last commit](https://github.com/CEED/libCEED/pull/995/commits/b70ae9d3e37855b20b547f7fe4ba07cd52b70e4b)?

Reply:
Great, modulo one comment I left there.

Reply:
Cool, just fixed renamed it.